src/engine/blst.rs

```diff
@@ -0,0 +1,366 @@
+        let commitment = backend.commit_to_poly(poly.clone()).unwrap();
+        let proof = backend.compute_proof_single(poly.clone(), x).unwrap();
+        let result = backend
+            .verify_proof_single(proof, x, y, commitment)
+            .unwrap();
+        assert!(result);
+    }
+
+    #[test]
+    fn test_arkworks_g1_serialize_deserialize() {
+        let cfg = crate::engine::backend::BackendConfig::new(Some(4), None);
+        let backend = BlstBackend::new(Some(cfg.clone()));
+        let g1 = backend.random_g1();
+        println!("g1: {:?}", g1);
+        let g1_bytes = g1.to_bytes();
+        let reserialized = FsG1::from_bytes(&g1_bytes);
+        assert_eq!(reserialized, Ok(g1));
+    }
+
+    // TODO: make proper test for fixed setup file and fixed secret
+    // #[test]
+    // #[tracing_test::traced_test]
+    // fn test_pipeline() {
+    //     let cfg = crate::engine::backend::BackendConfig::new(Some(4), Some("setup".to_string()));
+    //     let backend = BlstBackend::new(Some(cfg.clone()));
+    //
+    //     // Get hardcoded poly
+    //     let poly = backend
+    //         .parse_poly_from_str(
+    //             &TEST_POLY
+    //                 .iter()
+    //                 .map(|x| x.to_string())
+    //                 .collect::<Vec<String>>(),
+    //         )
+    //         .expect("Failed to parse poly");
+    //
+    //     // Get hardcoded point
+    //     let x = backend
+    //         .parse_point_from_str(TEST_POINT)
+    //         .expect("Failed to parse point");
+    //     println!("x: {:?}", hex::encode(x.to_bytes()));
+    //
+    //     // Evaluate poly at point and check against hardcoded value
+    //     let y = poly.eval(&x);
+    //     let expected = backend
+    //         .parse_point_from_str(TEST_EVAL)
+    //         .expect("Failed to parse point");
+    //     println!("y: {:?}", hex::encode(y.to_bytes()));
+    //     assert_eq!(y, expected);
+    //
+    //     // Commit to poly and check against hardcoded commitment
+    //     let commitment = backend
+    //         .commit_to_poly(poly.clone())
+    //         .expect("Failed to commit to poly");
+    //     println!("commitment hex: {:?}", hex::encode(commitment.to_bytes()));
+    //     assert_eq!(hex::encode(commitment.to_bytes()), EXPECTED_COMMITMENT);
+    //
+    //     // Compute proof and check against hardcoded proof
+    //     let proof = backend
+    //         .compute_proof_single(poly.clone(), x)
+    //         .expect("Failed to compute proof");
+    //     println!("proof hex: {:?}", hex::encode(proof.to_bytes()));
+    //     assert_eq!(hex::encode(proof.to_bytes()), EXPECTED_PROOF);
+    //
+    //     // Verify proof
+    //     let result = backend
+    //         .verify_proof_single(proof, x, y, commitment)
+    //         .expect("Failed to verify proof");
+    //     assert!(result);
+    // }
+
+    #[test]
+    #[tracing_test::traced_test]
+    fn write_setup() {
```

Comment:
Should likely be commented out or removed since we have a function for it now right?